### ISSUE TITLE
no object in Resource

### ISSUE THREAD

**Issue by zoran** [jsonapi-suite/jsonapi_compliable#10]
in a more complex scenario I want to access attributes of the @object in a Resource (just like in a Serializer). Is that possible somehow? 

```
class UserResource
  type :users
  
  has_many :addresses,
    resource: AdressResource,
    foreign_key: :abc,
    primary_key: :xyz,
    scope -> { Adress.where("zip LIKE ?", @object.zip) }
end
```

**Comment by richmolj** [comment]
@zoran I think this is probably best left to the lower-level DSL, which you can customize with a block. We wouldn't be dealing with `@object` at this point but `@objects` (because you could be hitting an index endpoint listing all users, then want all addresses for all users). That's available in the lower-level DSL:

```ruby
class UserResource
  type :users
  
  has_many :addresses, options do
    scope do |users|
      Address.where(["zip LIKE ?", users.map(&:zip))
    end
  end
end
```

Does that work for you? Note this will override any `scope` option you pass to `has_many`.

**Comment by zoran** [comment]
mhh, I get `missing keywords: scope` and I have to use an empty scope option:
```
has_many :addresses,
  scope: -> {} do
    scope do |users|
      Address.where(["zip LIKE ?", users.map(&:zip))
    end
  end
```

However, the options `foreign_key` and `primary_key` are ignored. Of course, it's possible to deal with it by change the `where` statement accordingly but it doesn't feel right.

**Comment by richmolj** [comment]
Yeah, I should definitely make it so you can pass EITHER `scope` key, OR a `scope` block. `primary_key/foreign_key` would still be used to assign the objects to each other, though.

After that I'd say the abstraction still looks correct to me. Would just making `scope` optional sound good to you?

**Comment by zoran** [comment]
Yes that would be great

**Comment by richmolj** [comment]
@zoran this is in master, I will release a new version tonight

**Comment by richmolj** [comment]
Fixed in 0.5.5
